src/pages/iou/MoneyRequestModal.js

```diff
@@ -0,0 +1,499 @@
+        alternateText: Str.isSMSLogin(personalDetails.login) ? Str.removeSMSDomain(personalDetails.login) : personalDetails.login,
+        icons: [{
+            source: ReportUtils.getAvatar(personalDetails.avatar, personalDetails.login),
+            name: personalDetails.login,
+            type: CONST.ICON_TYPE_AVATAR,
+        }],
+        keyForList: personalDetails.login,
+        payPalMeAddress: lodashGet(personalDetails, 'payPalMeAddress', ''),
+        phoneNumber: lodashGet(personalDetails, 'phoneNumber', ''),
+    }));
+
+    // Skip IOUParticipants step if participants are passed in
+    const steps = reportParticipants.length ? [Steps.IOUAmount, Steps.IOUConfirm] : [Steps.IOUAmount, Steps.IOUParticipants, Steps.IOUConfirm];
+
+    const prevCreatingIOUTransactionStatusRef = useRef();
+    const prevNetworkStatusRef = useRef();
+    const prevSelectedCurrencyCodeRef = useRef();
+
+    const [previousStepIndex, setPreviousStepIndex] = useState(0);
+    const [currentStepIndex, setCurrentStepIndex] = useState(0);
+    const [participants, setParticipants] = useState(participantsWithDetails);
+    const [amount, setAmount] = useState('');
+    const [comment, setComment] = useState('');
+
+    useEffect(() => {
+        PersonalDetails.openMoneyRequestModalPage();
+        IOU.setIOUSelectedCurrency(props.currentUserPersonalDetails.localCurrencyCode);
+    }, []);
+
+    useEffect(() => {
+        prevCreatingIOUTransactionStatusRef.current = lodashGet(props, 'iou.creatingIOUTransaction');
+    }, [props.iou]);
+
+    useEffect(() => {
+        prevNetworkStatusRef.current = props.network.isOffline;
+    }, [props.network.isOffline]);
+
+    useEffect(() => {
+        prevSelectedCurrencyCodeRef.current = lodashGet(props, 'iou.selectedCurrencyCode');
+    }, [props.iou.selectedCurrencyCode]);
+
+    useEffect(() => {
+        const isDoneCreatingIOUTransaction = prevCreatingIOUTransactionStatusRef.current && !lodashGet(props, 'iou.creatingIOUTransaction');
+
+        // User came back online, so we can try to create the IOU transaction again
+        if (prevNetworkStatusRef.current && !props.network.isOffline) {
+            PersonalDetails.openMoneyRequestModalPage();
+        }
+
+        // If the curency code is updated, also update the IOU selected currency
+        const selectedCurrencyCode = lodashGet(props, 'iou.selectedCurrencyCode');
+        if (prevSelectedCurrencyCodeRef.current !== selectedCurrencyCode) {
+            IOU.setIOUSelectedCurrency(selectedCurrencyCode);
+        }
+
+        if (!isDoneCreatingIOUTransaction) {
+            return;
+        }
+
+        // At this point, we're done creating the IOU transaction, but we need to check if it was successful
+        if (lodashGet(props, 'iou.error') === true) {
+            setCurrentStepIndex(0);
+        } else {
+            Navigation.dismissModal();
+        }
+    }, [props.iou, props.network.isOffline]);
```

Comment:
Might be better as two separate `useEffect()` (ignore if we take my advice above and create a custom hook for the network stuff)